diff --git a/ocs2_core/include/ocs2_core/integration/TrapezoidalIntegration.h b/ocs2_core/include/ocs2_core/integration/TrapezoidalIntegration.h
index 37088b1..5493689 100644
--- a/ocs2_core/include/ocs2_core/integration/TrapezoidalIntegration.h
+++ b/ocs2_core/include/ocs2_core/integration/TrapezoidalIntegration.h
@@ -48,7 +48,7 @@ VALUE_T trapezoidalIntegration(const std::vector<SCALAR_T>& timeTrajectory, cons
     return initialValue;
   }
 
-  for (size_t k = 1; k < timeTrajectory.size(); k++) {
+  for (std::size_t k = 1; k < timeTrajectory.size(); k++) {
     auto temp = valueTrajectory[k - 1] + valueTrajectory[k];
     temp *= (0.5 * (timeTrajectory[k] - timeTrajectory[k - 1]));
     initialValue += temp;
diff --git a/ocs2_robotic_examples/ocs2_legged_robot/src/common/ModelSettings.cpp b/ocs2_robotic_examples/ocs2_legged_robot/src/common/ModelSettings.cpp
index 669cd5c..41dd4c9 100644
--- a/ocs2_robotic_examples/ocs2_legged_robot/src/common/ModelSettings.cpp
+++ b/ocs2_robotic_examples/ocs2_legged_robot/src/common/ModelSettings.cpp
@@ -43,6 +43,10 @@ ModelSettings loadModelSettings(const std::string& filename, const std::string&
   loadData::loadPtreeValue(pt, modelSettings.recompileLibrariesCppAd, fieldName + ".recompileLibrariesCppAd", verbose);
   loadData::loadPtreeValue(pt, modelSettings.modelFolderCppAd, fieldName + ".modelFolderCppAd", verbose);
 
+  loadData::loadStdVector(filename, fieldName + ".jointNames", modelSettings.jointNames, verbose);
+  loadData::loadStdVector(filename, fieldName + ".contactNames3DoF", modelSettings.contactNames3DoF, verbose);
+  loadData::loadStdVector(filename, fieldName + ".contactNames6DoF", modelSettings.contactNames6DoF, verbose);
+
   if (verbose) {
     std::cerr << " #### =============================================================================" << std::endl;
   }
diff --git a/ocs2_sqp/blasfeo_catkin/CMakeLists.txt b/ocs2_sqp/blasfeo_catkin/CMakeLists.txt
index 01a6db6..68a991e 100644
--- a/ocs2_sqp/blasfeo_catkin/CMakeLists.txt
+++ b/ocs2_sqp/blasfeo_catkin/CMakeLists.txt
@@ -6,6 +6,10 @@ include(FetchContent)
 # BLASFEO settings
 set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries" FORCE)
-set(TARGET X64_AUTOMATIC CACHE STRING "Target architecture" FORCE)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
+  set(TARGET ARMV8A_ARM_CORTEX_A57 CACHE STRING "Target architecture" FORCE)
+else()
+  set(TARGET X64_AUTOMATIC CACHE STRING "Target architecture" FORCE)
+endif()
 set(BLASFEO_EXAMPLES OFF CACHE BOOL "Examples enabled" FORCE)
 
 FetchContent_Declare(blasfeodownload
